planner/cascades/optimize.go

```diff
@@ -343,12 +343,14 @@ func (opt *Optimizer) implGroupExpr(cur *memo.GroupExpr, reqPhysProp *property.P
 		if !rule.Match(cur, reqPhysProp) {
 			continue
 		}
-		impl, err := rule.OnImplement(cur, reqPhysProp)
+		implementations, err := rule.OnImplement(cur, reqPhysProp)
```

Comment:
The two names `impls`, `implementations` are confusing. Do we have another choice?